Feature: Handle Arabic

1saifj opened the pull request:
This Close #89

1saifj committed:
feat(supports-arabic): support comprehensive Arabic texts

1saifj committed:
add one more test case

1saifj committed:
add check for arabic chars and special cases

1saifj committed:
handle more speicla cases

1saifj committed:
add more test cases

1saifj commented:
@antonlindstrom @navruzm Your review please

1saifj commented:
@antonlindstrom @navruzm @matrixik @kvalev ur review please

matrixik commented:
Hello, thank you for this change.

Please revert change to `BenchmarkMakeShortSymbols` (there is already test for simple text `BenchmarkMakeShortAscii`).

Also please lower the number of examples in `doc.go` to 1 example (best would be the trickiest one, it's there to show it's possible, not to show all abilities for one language).

From my google search Arabic doesn't have upper letters but could you add one test for it like:
```
{"ar", "مرحبا بالعالم", "mrhba-balalm", false},
```

1saifj commented:
Ok let me do it and sorry for the long time delay...

1saifj committed:
Address PR review feedback

- Revert BenchmarkMakeShortSymbols to use original special symbols string
- Reduce Arabic examples in doc.go to 1 (trickiest case with plurals)
- Add Arabic test case with Lowercase=false
